src/strategies/hedgey-multi/index.ts

```diff
@@ -0,0 +1,192 @@
+  if (contractDetails.length > MAX_CONTRACTS) {
+    throw new Error(`Max number (${MAX_CONTRACTS}) of contracts exceeded`);
+  }
+  const balanceOfMulti = new Multicaller(network, provider, abis.NFT, {
+    blockTag
+  });
+
+  contractDetails.forEach((contractDetail) => {
+    addresses.forEach((address: string) => {
+      balanceOfMulti.call(
+        `${contractDetail.address}/${address}`,
+        contractDetail.address,
+        'balanceOf',
+        [address]
+      );
+    });
+    if (contractDetail.contractType === ContractType.TokenInfusedNFT) {
+      balanceOfMulti.call(
+        `${contractDetail.address}/token`,
+        contractDetail.address,
+        'token'
+      );
+    }
+  });
+
+  const balanceOfResult = await balanceOfMulti.execute();
+  const nftHolderMulti = new Multicaller(network, provider, abis.NFT, {
+    blockTag
+  });
+
+  contractDetails.forEach((contractDetail) => {
+    addresses.forEach((address: string) => {
+      const balance = balanceOfResult[`${contractDetail.address}/${address}`];
+      for (let index = 0; index < balance; index++) {
+        nftHolderMulti.call(
+          `${contractDetail.contractType}/${contractDetail.address}/${address}/${index}`,
+          contractDetail.address,
+          'tokenOfOwnerByIndex',
+          [address, String(index)]
+        );
+      }
+    });
+  });
+
+  const nftHolders = await nftHolderMulti.execute();
+
+  const nftDealsMulti = new Multicaller(network, provider, abis.NFT, {
+    blockTag
+  });
+
+  const tiNFTDealsMulti = new Multicaller(
+    network,
+    provider,
+    abis.TokenInfusedNFT,
+    { blockTag }
+  );
+
+  for (const [path, nftId] of Object.entries(nftHolders)) {
+    const [contractType, contractAddress, address] = path.split('/');
+    switch (contractType) {
+      case ContractType.NFT:
+        nftDealsMulti.call(
+          `${contractAddress}/${address}`,
+          contractAddress,
+          'futures',
+          [nftId]
+        );
+        break;
+      case ContractType.TokenInfusedNFT:
+        tiNFTDealsMulti.call(
+          `${contractAddress}/${address}`,
+          contractAddress,
+          'futures',
+          [nftId]
+        );
+        break;
+    }
+  }
+
+  const nftDeals = await nftDealsMulti.execute();
+  const tiNFTDeals = await tiNFTDealsMulti.execute();
+
+  const votes = {};
+
+  for (const [path, deal] of Object.entries<any>(nftDeals)) {
+    const [contractAddress, address] = path.split('/');
+    const contractDetail = contractDetails.find(
+      (element) => element.address === contractAddress
+    );
+
+    if (!contractDetail) continue;
+    if (!compareAddresses(deal.token, contractDetail.token)) continue;
+
+    const amount = BigNumber.from(deal.amount).div(
+      BigNumber.from(10).pow(contractDetail.decimal)
+    );
+
+    let score = amount.toNumber();
+
+    if (contractDetail.multiplyer) {
+      const multiplyer = eval(contractDetail.multiplyer);
+      score = multiplyer(score, deal);
```

Comment:
Hey @bark4mark Sorry for not noticing this earlier, But we can accept `eval` , There is a chance of security risk. can you try passing the multiplier value directly?

Reply:
I have removed the eval